### PR TITLE
Message status on all types of messages.

### LINKED ISSUE #12944: Add message statuses to all types of messages
# Feature Issue

### User Story

As a user, I want to see that all types of messages (stickers, commands, pictures, audio) are sent.
cc @hesterbruikman 
### Description
Currently only for text messages can see statuses (sent / delivered / pending)
IMAGE 2021-12-21 13:17:18

### Acceptance Criteria
Can see statuses for all types of messages.

### PR DESCRIPTION
**Pull request by jo-mut**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #12944

### Summary

[comment]: # This PR adds message status on images and sticker messages as the other types of message already have this implementation
...

Documentation change PR (review please): https://github.com/status-im/status.im/pull/xxx

#### Platforms

- Android
- iOS
- macOS
- Linux
- Windows

##### Functional

- 1-1 chats
- public chats
- group chats
- wallet / transactions
- dapps / app browsing
- account recovery
- new account
- user profile updates
- networks
- mailservers
- fleet
- bootnodes

##### Non-functional

- battery performance
- CPU performance / speed of the app
- network consumption

### Steps to test

- Open Status
- ...
- Step 3, etc.

status: ready

### CODE REVIEW

**Comment by jo-mut** [comment]
I hadnt tested this feature on ios. I can see its not outputting a
successful build on ios. Let me work on it

File: src/status_im/signals/core.cljs [hunk 1]
```diff
@@ -67,6 +67,8 @@
       "envelope.expired"   (transport.message/update-envelopes-status cofx (:ids (js->clj event-js :keywordize-keys true)) :not-sent)
       "message.delivered"  (let [{:keys [chatID messageID]} (js->clj event-js :keywordize-keys true)]
                              (models.message/update-db-message-status cofx chatID messageID :delivered))
+     ;; "message.seen"       (let [{:keys [chatID messageID]} (js->clj event-js :keywordize-keys true)]
```
Comment (flexsurfer) [inline src/status_im/signals/core.cljs]: could you elaborate please?

**Review by flexsurfer — COMMENTED** [review]
is this PR ready ?

File: src/status_im/signals/core.cljs (hunk 1 above)
Comment (jo-mut) [reply src/status_im/signals/core.cljs]: I forgot also to remove this line. I had been working on the develop branch while working on another feature. I sorted that let me remove this line of code

File: src/status_im/subs.cljs [hunk 2]
```diff
@@ -2261,6 +2261,12 @@
  (fn [multiaccount]
    (get multiaccount :profile-pictures-visibility)))
 
+(re-frame/reg-sub
```
Comment (flexsurfer) [inline src/status_im/subs.cljs]: this also doesn't look related

**Comment by flexsurfer** [comment]
@jo-mut could you please make sure this PR includes only related changes, thanks

File: src/status_im/subs.cljs (hunk 2 above)
Comment (jo-mut) [reply src/status_im/subs.cljs]: Am removing the unrelated code in this PR. It was my mistake to work on the same branch. Should be okay after.

File: src/status_im/signals/core.cljs [hunk 3]
```diff
@@ -75,4 +75,4 @@
       "wallet" (ethereum.subscriptions/new-wallet-event cofx (js->clj event-js :keywordize-keys true))
       "local-notifications" (local-notifications/process cofx (js->clj event-js :keywordize-keys true))
       "community.found" (link.preview/cache-community-preview-data (js->clj event-js :keywordize-keys true))
-      (log/debug "Event " type " not handled"))))
+      (log/debug "Event " type " not handled "))))
```
Comment (flexsurfer) [inline src/status_im/signals/core.cljs]: could you please revert this change

Comment (jo-mut) [reply src/status_im/signals/core.cljs]: yeah got it. thanks

**Comment by flexsurfer** [comment]
hey @jo-mut is this PR ready ?

**Comment by jo-mut** [comment]
Yeah it is

**Comment by flexsurfer** [comment]
what about audio message ?

**Comment by jo-mut** [comment]
I had seen that it had already been implemented maybe i missed it. Let me see and work on it.

**Comment by churik** [comment]
Please, make sure your branch is rebased on current `develop`. Thanks!

**Comment by jo-mut** [comment]
Okay

**Comment by pavloburykh** [comment]
Thanks for PR.  Tested on Android build. 

PR issues: 

### 1. Unable to install last successful IOS build, cause it is not available anymore. Rebuilding results in error: 
```
 /Users/jenkins/workspace/status-react_prs_ios_PR-12962@tmp/durable-306548a9/script.sh: line 3: /Users/jenkins/.nix-profile/etc/profile.d/nix.sh: No such file or directory
```
https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-12962/


### 2.  Images and sticker messages still don't have statuses  (sent / delivered / pending)

photo_2022-01-17 15 09 39

https://user-images.githubusercontent.com/97245802/149777269-5f09780a-988a-46ee-9294-750a62306249.mp4

[Status-debug-logs-messages-status.zip](https://github.com/status-im/status-react/files/7882116/Status-debug-logs-messages-status.zip)

### 3. Audio messages stuck in pending status after sending. I fact audio message can be already sent and delivered but pending status still displayed. I t can be observed on screenshot and video given above.

**Comment by jo-mut** [comment]
Message status on stickers is actually okay the issue with that is the
color of the icon is basically the same with the background color which is
white which is why they are rendered invisible

**Comment by jo-mut** [comment]
I have seen the issue with audio messages, could be a bug. I will consult
on this but eventually it still appears to update the status as having been
sent when i tested it.

**Comment by pavloburykh** [comment]
> 

In my case - pending status of audio message has not been updated most of the times. Although once it still was updated to delivered status. Hope that logs attached here https://github.com/status-im/status-react/pull/12962#issuecomment-1014542331 will help somehow. 

As for the STRs, for now I don't have any special ones. I'm just sending audio messages in 1-1 or group chats and they stuck in pending status most of the times.

**Comment by jo-mut** [comment]
What I realized is the for the audio messages the status is updated to sent when the application goes to the background. This is a bug. I will have a look

**Comment by pavloburykh** [comment]
> Message status on stickers is actually okay the issue with that is the
color of the icon is basically the same with the background color which is
white which is why they are rendered invisible

@pedro-et hi! Need your advice on some design issues related to this PR, will appreciate if you can help. 

## Problems: 

1. Currently, status icons (such as pending/sent/delivered) are not visible for stickers messages both for light and dark modes as icons have the same colour as the background (white for light theme and black for dark). Screenshots attached. Probably we need to change status icons' colour in order to make them visible on white and dark backgrounds. 

photo_2022-01-17 16 50 25
photo_2022-01-17 16 50 33

2. In case of image messages - status icons are located exactly on the image which eventually often results in icons to be invisible on some dark or white images. Screenshots attached. Maybe it would be better to locate status icon to the right of the image (just like in case with sticker messages) instead of locating them on the image itself? 

photo_2022-01-17 16 51 15
photo_2022-01-17 16 51 22

**Comment by pedro-et** [comment]
You can find the design files for this here:
https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=17651%3A122

**Comment by pavloburykh** [comment]
> You can find the design files for this here: https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=17651%3A122

thanks a lot for helping out @pedro-et 
@jo-mut could you please use given designs to fix current issues?

**Comment by jo-mut** [comment]
@pavloburykh cool. I will fix the issues following the designs

**Comment by jo-mut** [comment]
@pavloburykh I have a feature request, could we also have a design for a captioned image. currently when you send an image message with a caption the caption and the image are both appearing as too separate messages. What do you think?

**Comment by du82** [comment]
@jo-mut I think we should have captions just like how WeChat does them. Tap the image and the caption is in the image viewer.

Screenshot_20220123-082143_WeChat

**Comment by jo-mut** [comment]
Yeah something like that  at the moment this is missing

**Comment by jo-mut** [comment]
Hi @pavloburykh. I wanted to confirm something with the design team on the following commit. This commit (
https://github.com/status-im/status-react/commit/f3533c5f99ab6a64447428db872f7cafbd9cc7ee ) removed the time stamp from the chat bubbles but following the it re-adds the timestamp to the chat bubbles. I just want to confirm if we should still proceed with the designs that if this hadnt been to your attention

**Comment by pavloburykh** [comment]
> @pavloburykh I have a feature request, could we also have a design for a captioned image. currently when you send an image message with a caption the caption and the image are both appearing as too separate messages. What do you think?

@jo-mut yeah, I agree that it would be a good improvement, although I would implement it within a separate feature request not within current issue. @hesterbruikman @pedro-et whats your opinion about that? Should we implement this feature or there are some reasons why we haven't done it before?

**Comment by pedro-et** [comment]
@pavloburykh I think its a good feature but as will be resigning the chat experience as a whole, maybe its worth the wait! @hesterbruikman @John-44 what do you think?

**Comment by John-44** [comment]
> @pavloburykh I think its a good feature but as will be resigning the chat experience as a whole, maybe its worth the wait! @hesterbruikman @John-44 what do you think?

I'm *hoping* (but not set in stone by any means yet - still needs to be discussed and agreed with many parties) that we will be able to kick of the development of the Mobile redesign in 3 to 4 weeks.  In order to try achieve this timeline, mobile design needs to be 100% focused on the new design, and 4 weeks isn't long to wait for the new designs.  So if new design work is required here, I would lean towards doing it in the context of the new mobile design, not the current mobile design.

**Comment by pavloburykh** [comment]
@pedro-et @John-44 thanks for your answers!

**Comment by jo-mut** [comment]
Should we close this issue since the current designs don`t really work as desired with the message status and like @John-44 said its rather better to lean towards the new mobile design than the current one

**Comment by du82** [comment]
> @pavloburykh I think its a good feature but as will be resigning the chat experience as a whole, maybe its worth the wait! @hesterbruikman @John-44 what do you think?

Could I see Figma designs? I really like the current chat design, I just hate the way navigation is done throughout the entire app

**Comment by pavloburykh** [comment]
> > @pavloburykh I think its a good feature but as will be resigning the chat experience as a whole, maybe its worth the wait! @hesterbruikman @John-44 what do you think?
> 
> Could I see Figma designs? I really like the current chat design, I just hate the way navigation is done throughout the entire app

Hey @ducheng0 Thanks for reaching out! You can find designs here https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=17651%3A122

**Comment by du82** [comment]
> > > @pavloburykh I think its a good feature but as will be resigning the chat experience as a whole, maybe its worth the wait! @hesterbruikman @John-44 what do you think?
> > 
> > 
> > Could I see Figma designs? I really like the current chat design, I just hate the way navigation is done throughout the entire app
> 
> Hey @ducheng0 Thanks for reaching out! You can find designs here https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=17651%3A122

That looks like the current design. Is it a redo of the backend stuff? or will the UI also change later on?

**Comment by pavloburykh** [comment]
> > > > @pavloburykh I think its a good feature but as will be resigning the chat experience as a whole, maybe its worth the wait! @hesterbruikman @John-44 what do you think?
> > > 
> > > 
> > > Could I see Figma designs? I really like the current chat design, I just hate the way navigation is done throughout the entire app
> > 
> > 
> > Hey @ducheng0 Thanks for reaching out! You can find designs here https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=17651%3A122
> 
> That looks like the current design. Is it a redo of the backend stuff? or will the UI also change later on?

I guess @pedro-et can provide more accurate information about the designs.
